chutney/server/src/test/java/com/chutneytesting/execution/domain/PurgeServiceTest.java

```diff
@@ -86,6 +90,37 @@ void no_purge_when_limit_not_exceeded() {
                     assertThat(report.scenariosExecutionsIds()).isEmpty();
                 }
 
+                @Test
+                void no_purge_when_only_today_executions_even_if_limit_exceeded() {
```

Comment:
I would add a new nested class for explicitly say that there is a new rule : excluding executions from purge if happened within a configuration parameter.